src/ucp/tag/offload/rndv.c

```diff
@@ -0,0 +1,309 @@
+                                            cap.tag.rndv.max_iov),
+        .super.hdr_size      = 0,
+        .super.send_op       = UCT_EP_OP_RNDV_ZCOPY,
+        .super.memtype_op    = UCT_EP_OP_LAST,
+        .super.flags         = UCP_PROTO_COMMON_INIT_FLAG_RESPONSE,
+        .remote_op_id        = UCP_OP_ID_RNDV_RECV_OFFLOAD,
+        .unpack_time         = UCS_LINEAR_FUNC_ZERO,
+        .perf_bias           = context->config.ext.rndv_perf_diff / 100.0,
+        .mem_info.type       = init_params->select_param->mem_type,
+        .mem_info.sys_dev    = init_params->select_param->sys_dev
+    };
+    ucp_lane_index_t lane;
+    ucp_md_index_t md_index;
+
+    if (!ucp_proto_tag_check_op_id(init_params) ||
+        !ucp_ep_config_key_has_tag_lane(init_params->ep_config_key) ||
+        (init_params->select_param->dt_class != UCP_DATATYPE_CONTIG) ||
+        !ucp_proto_tag_offload_rndv_supported(init_params, &params)) {
+        return UCS_ERR_UNSUPPORTED;
+    }
+
+    /* Multirail is not supported with tag offload rndv, need to register
+     * memory on tag lane only. */
+    lane               = init_params->ep_config_key->tag_lane;
+    md_index           = ucp_proto_common_get_md_index(init_params, lane);
+    rpriv->md_map      = UCS_BIT(md_index);
+    rpriv->sys_dev_map = 0;
+
+    return ucp_proto_rndv_ctrl_init(&params, lane);
+}
+
+static void
+ucp_proto_tag_rndv_offload_completion(uct_completion_t *uct_comp)
+{
+    ucp_request_t *req = ucs_container_of(uct_comp, ucp_request_t,
+                                          send.state.uct_comp);
+
+    ucp_send_request_id_release(req);
+    ucp_proto_request_zcopy_complete(req, uct_comp->status);
+}
+
+static UCS_F_ALWAYS_INLINE ucs_status_t
+ucp_proto_tag_rndv_offload_rts_request_init(ucp_request_t *req)
+{
+    const ucp_proto_rndv_ctrl_priv_t *rpriv = req->send.proto_config->priv;
+    ucs_status_t status;
+
+    if (req->flags & UCP_REQUEST_FLAG_PROTO_INITIALIZED) {
+        return UCS_OK;
+    }
+
+    status = ucp_ep_resolve_remote_id(req->send.ep, rpriv->lane);
+    if (status != UCS_OK) {
+        return status;
+    }
+
+    status = ucp_proto_request_zcopy_init(
+                 req, rpriv->md_map, ucp_proto_tag_rndv_offload_completion,
+                 UCT_MD_MEM_ACCESS_RMA | UCT_MD_MEM_FLAG_HIDE_ERRORS,
+                 UCP_DT_MASK_ALL);
+    if (status != UCS_OK) {
+        return status;
+    }
+
+    ucp_send_request_id_alloc(req);
+    req->flags |= UCP_REQUEST_FLAG_PROTO_INITIALIZED;
+
+    return UCS_OK;
+}
+
+UCS_PROFILE_FUNC(ucs_status_t, ucp_proto_tag_rndv_offload_progress, (self),
+                 uct_pending_req_t *self)
+{
+    ucp_request_t *req = ucs_container_of(self, ucp_request_t, send.uct);
+    ucp_ep_h ep        = req->send.ep;
+    size_t max_iov     = ucp_ep_config(ep)->tag.offload.max_rndv_iov;
+    uct_iov_t *iov     = ucs_alloca(max_iov * sizeof(uct_iov_t));
+    const ucp_proto_rndv_ctrl_priv_t *rpriv;
```

Comment:
we only use `lane` of `rpriv`.
```suggestion
    ucp_lane_index_t lane;
...
    lane = req->send.proto_config->priv->lane;
...
    rndv_hdr.md_index = ucp_ep_md_index(ep, lane);
...
    rndv_op = uct_ep_tag_rndv_zcopy(ep->uct_eps[lane],
...
    return ucp_proto_single_status_handle(req, 1, NULL, lane,
```